packages/runtime/test/integration/SDKTxnRuntime.ts

```diff
@@ -0,0 +1,148 @@
+import { types } from "@algo-builder/web";
+import { mkTransaction } from "@algo-builder/web/build/lib/txn";
+import { ASADef } from "@algo-builder/web/build/types";
+import algosdk, { makeApplicationCreateTxn, makeAssetCreateTxnWithSuggestedParams, OnApplicationComplete, Transaction } from "algosdk";
+import { assert } from "chai";
+
+import { getProgram, loadASAFile } from "../../src";
+import { AccountStore } from "../../src/account";
+import { convertToString } from "../../src/lib/parsing";
+import { mockSuggestedParams } from "../../src/mock/tx";
+import { Runtime } from "../../src/runtime";
+import { AccountStoreI, TxReceipt } from "../../src/types";
+import { useFixture } from "../helpers/integration";
+
+const minBalance = BigInt(1e7);
+
+// TODO: add more test for all transaction types.
+describe("Execute SDK transaction", function () {
+  const fee = 1000;
+
+  let alice: AccountStoreI;
+  let smith: AccountStoreI;
+  let runtime: Runtime;
+
+  let execParams: types.ExecParams;
+
+  function mkTransactionAndSign (runtime: Runtime, execParams: types.ExecParams): types.TransactionAndSign {
+    const suggestedParams = mockSuggestedParams(execParams.payFlags, runtime.getRound());
+    const transaction = mkTransaction(execParams, suggestedParams) as any;
+    let sign: types.Sign;
+
+    // extract `sign` from execParams
+    if (execParams.sign === types.SignType.SecretKey) {
+      sign = {
+        sign: execParams.sign,
+        fromAccount: execParams.fromAccount
+      };
+    } else {
+      sign = {
+        sign: execParams.sign,
+        lsig: execParams.lsig,
+        fromAccountAddr: execParams.fromAccountAddr
+      };
+    }
+
+    // inject approval and clear program in string format to transaction object.
+    // TODO: Should we create disassemble method to convert Uint8Array program format to string???
+    if (execParams.type === types.TransactionType.DeployApp) {
+      transaction.approvalProgram = execParams.approvalProgram;
+      transaction.clearProgram = execParams.clearProgram;
+    }
+    return {
+      transaction,
+      sign
+    };
+  }
+
+  this.beforeEach(() => {
+    alice = new AccountStore(minBalance * 10n);
+    smith = new AccountStore(minBalance * 10n);
+    runtime = new Runtime([alice, smith]);
+  });
+
+  describe("ASA transaction", function () {
+    useFixture('asa-check');
+    it("Deploy ASA transaction", () => {
+      const asaName = 'gold';
+      const asaDef = runtime.loadedAssetsDefs[asaName];
+
+      execParams = {
+        sign: types.SignType.SecretKey,
+        fromAccount: alice.account,
+        type: types.TransactionType.DeployASA,
+        asaName,
+        asaDef,
+        payFlags: {
+          totalFee: fee
+        }
+      };
+
+      const txAndSign = mkTransactionAndSign(runtime, execParams);
```

Comment:
How about adding a test directly using `SDK` create methods (eg. `mkAssetTransferTransactionFromSuggestedParams`). I think user will directly use that function and call runtime's executeTx, not going through `execParams` --> `mkTransactionAndSign`. 

But if that changes too much of this test, let's leave as it is.

Reply:
> How about adding a test directly using `SDK` create methods (eg. `mkAssetTransferTransactionFromSuggestedParams`). I think user will directly use that function and call runtime's executeTx, not going through `execParams` --> `mkTransactionAndSign`.
> 
> But if that changes too much of this test, let's leave as it is.

First time, I also have same approach like you(use mkAssetTransferTransactionFromSuggestedParams, makeApplicationCreateTxn)  However, I see in method [mkTransaction](https://github.com/scale-it/algo-builder/blob/0650e5283ddf92633dab54ba8888034c0bd56cef/packages/web/src/lib/txn.ts#L57) in `algob/web` we ready create SDK transaction. My idea is proof we can run SDK transaction in Runtime and reuse mkTransaction method help me create SDK Txn easier.

Reply:
OK